File: Plugin/src/main/java/xyz/oribuin/skyblock/world/LayeredChunkGenerator.java
package xyz.oribuin.skyblock.world;

import org.bukkit.Material;
import org.bukkit.generator.ChunkGenerator;
import org.bukkit.generator.WorldInfo;
import org.jetbrains.annotations.NotNull;

import java.util.*;

public class LayeredChunkGenerator extends ChunkGenerator {

    @Override
    public void generateSurface(@NotNull WorldInfo worldInfo, @NotNull Random random, int chunkX, int chunkZ, @NotNull ChunkGenerator.ChunkData chunkData) {
        chunkData.setRegion(0, worldInfo.getMinHeight(), 0, 16, worldInfo.getMaxHeight(), 16, Material.AIR);
    }

    @Override
    public void generateCaves(@NotNull WorldInfo worldInfo, @NotNull Random random, int chunkX, int chunkZ, @NotNull ChunkGenerator.ChunkData chunkData) {
        chunkData.setRegion(0, worldInfo.getMinHeight(), 0, 16, worldInfo.getMaxHeight(), 16, Material.AIR);
    }

    @Override
    public void generateBedrock(@NotNull WorldInfo worldInfo, @NotNull Random random, int chunkX, int chunkZ, @NotNull ChunkGenerator.ChunkData chunkData) {
        chunkData.setRegion(0, worldInfo.getMinHeight(), 0, 16, worldInfo.getMaxHeight(), 16, Material.AIR);
    }

    @Override
    public void generateNoise(@NotNull WorldInfo worldInfo, @NotNull Random random, int chunkX, int chunkZ, @NotNull ChunkGenerator.ChunkData chunkData) {
        chunkData.setRegion(0, worldInfo.getMinHeight(), 0, 16, worldInfo.getMaxHeight(), 16, Material.AIR);
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/world/LayeredChunkGenerator.kt
package xyz.oribuin.skyblock.world

import java.util.*
import org.bukkit.Material
import org.bukkit.generator.ChunkGenerator
import org.bukkit.generator.WorldInfo

class LayeredChunkGenerator : ChunkGenerator() {

    /// TODO: Add support for biomes
    override fun generateSurface(worldInfo: WorldInfo, random: Random, x: Int, z: Int, chunkData: ChunkData) =
        chunkData.setRegion(0, worldInfo.minHeight, 0, 16, worldInfo.maxHeight, 16, Material.AIR)

    override fun generateCaves(worldInfo: WorldInfo, random: Random, chunkX: Int, chunkZ: Int, chunkData: ChunkData) =
        chunkData.setRegion(0, worldInfo.minHeight, 0, 16, worldInfo.maxHeight, 16, Material.AIR)

    override fun generateBedrock(worldInfo: WorldInfo, random: Random, chunkX: Int, chunkZ: Int, chunkData: ChunkData) =
        chunkData.setRegion(0, worldInfo.minHeight, 0, 16, worldInfo.maxHeight, 16, Material.AIR)

    override fun generateNoise(worldInfo: WorldInfo, random: Random, chunkX: Int, chunkZ: Int, chunkData: ChunkData) =
        chunkData.setRegion(0, worldInfo.minHeight, 0, 16, worldInfo.maxHeight, 16, Material.AIR)
}